publications/tasks.py

```diff
@@ -53,68 +55,100 @@ def extract_timeperiod_from_html(content):
     # returning arrays for array field in DB
     return [period[0]], [period[1]]
 
-def parse_oai_xml_and_save_publications(content):
+def parse_oai_xml_and_save_publications(content, event):
+
     DOMTree = xml.dom.minidom.parseString(content)
-    collection = DOMTree.documentElement # pass DOMTree as argument
-    articles = collection.getElementsByTagName("dc:identifier")
-    articles_count_in_journal = len(articles)
-    for i in range(articles_count_in_journal):
-        identifier = collection.getElementsByTagName("dc:identifier")
-        identifier_value = identifier[i].firstChild.nodeValue
-        if identifier_value.startswith('http'):
+    collection = DOMTree.documentElement
 
-            with requests.get(identifier_value) as response:
-                soup = BeautifulSoup(response.content, 'html.parser')
+    records = collection.getElementsByTagName("record")
 
-                geom_object = extract_geometry_from_html(soup)
-                period_start, period_end = extract_timeperiod_from_html(soup)
+    if not records:
+        logger.warning("No articles found in OAI-PMH response!")
+        return
+
+    existing_urls = set(Publication.objects.values_list('url', flat=True))  # ✅ Cache existing URLs
+    existing_dois = set(Publication.objects.exclude(doi__isnull=True).values_list('doi', flat=True))  # ✅ Cache existing DOIs
+
+    for record in records:
+        try:
+            def get_text(tag_name):
+                nodes = record.getElementsByTagName(tag_name)
+                return nodes[0].firstChild.nodeValue.strip() if nodes and nodes[0].firstChild else None
+
+            identifier_value = get_text("dc:identifier")
+            title_value = get_text("dc:title")
+            abstract_text = get_text("dc:description")
+            journal_value = get_text("dc:publisher")
+            date_value = get_text("dc:date")
+
+            doi_nodes = record.getElementsByTagName("dc:identifier")
+            doi_text = next((node.firstChild.nodeValue.strip() for node in doi_nodes if "doi.org" in node.firstChild.nodeValue), None)
```

Comment:
This is not enough to identify DOIs, because journals may not use the full URL, see the `dc:identifier` in view-source:https://www.jvolcanica.org/ojs/index.php/volcanica/oai?verb=ListRecords&metadataPrefix=oai_dc

Here is a recommendation for a regex: https://stackoverflow.com/a/48524047 (please check comments and other answers before deciding which one to use.